portal-api/src/utils/date.js

```diff
@@ -1,26 +1,72 @@
-const { addBusinessDays, format, isSameDay } = require('date-fns');
+const { addBusinessDays, format, isSameDay, isWeekend, startOfMonth } = require('date-fns');
 
 /**
- * Adds business days to the given 'fromDate' also taking into account the given 'holidays'
- * @param {Date} fromDate - the date from which to add the business days
- * @param {number} numberOfBusinessDays - the number of business days to add to the 'fromDate'
- * @param {Date[]} holidays - an array of holidays to take into account
+ * @param {Date} date
+ * @param {Date[]} holidays
+ * @returns {boolean}
+ */
+const isHoliday = (date, holidays) => holidays.some((holiday) => isSameDay(date, holiday));
+
+/**
+ * @param {Date} date
+ * @param {Date[]} holidays
  * @returns {Date}
  */
-const addBusinessDaysWithHolidays = (fromDate, numberOfBusinessDays, holidays) => {
-  let result = fromDate;
+const addOneBusinessDayWithHolidays = (date, holidays) => {
+  let nextBusinessDay = addBusinessDays(date, 1);
+  while (isHoliday(nextBusinessDay, holidays)) {
+    nextBusinessDay = addBusinessDays(nextBusinessDay, 1);
+  }
+  return nextBusinessDay;
+};
 
-  const isHoliday = (date) => holidays.some((holiday) => isSameDay(date, holiday));
+/**
+ * Given a date in the month and an array of holidays, this
+ * returns the first business day of the month in the year
+ * provided. For example, if the 1st of the month is a Monday
+ * and that day is not a holiday, it will return the current
+ * date. If the 1st is a Saturday, and the next Monday is not
+ * a holiday, it will return the next Monday.
+ * @param {Date} dateInMonth - A date in the required month
+ * @param {Date[]} holidays - A list of dates which should be excluded as holidays
+ * @returns {Date}
+ */
+const getFirstBusinessDayOfMonth = (dateInMonth, holidays) => {
+  let firstBusinessDay = startOfMonth(dateInMonth);
+  if (isWeekend(firstBusinessDay)) {
+    firstBusinessDay = addOneBusinessDayWithHolidays(firstBusinessDay, holidays);
+  }
+  while (isHoliday(firstBusinessDay, holidays)) {
+    firstBusinessDay = addOneBusinessDayWithHolidays(firstBusinessDay, holidays);
+  }
+  return firstBusinessDay;
+};
```

Comment:
I don't think we need the while loop if we're using the `addOneBusinessDayWithHolidays` function? Could this just be
```
const getFirstBusinessDayOfMonth = (dateInMonth, holidays) => {
  const firstDayOfMonth = startOfMonth(dateInMonth);

  if (isWeekend(firstDayOfMonth) || isHoliday(firstDayOfMonth, holidays)) {
    return addOneBusinessDayWithHolidays(firstDayOfMonth, holidays);
  }

  return firstDayOfMonth;
};
```
?